File: src/main/java/work/lclpnet/ap2/game/cozy_campfire/CozyCampfireMiniGame.java
package work.lclpnet.ap2.game.cozy_campfire;

import net.minecraft.item.ItemStack;
import net.minecraft.item.Items;
import net.minecraft.util.Identifier;
import work.lclpnet.ap2.api.game.*;
import work.lclpnet.ap2.base.ApConstants;
import work.lclpnet.ap2.base.ArcadeParty;

public class CozyCampfireMiniGame implements MiniGame {

    @Override
    public Identifier getId() {
        return ArcadeParty.identifier("cozy_campfire");
    }

    @Override
    public GameType getType() {
        return GameType.TEAM;
    }

    @Override
    public String getAuthor() {
        return ApConstants.PERSON_LCLP;
    }

    @Override
    public ItemStack getIcon() {
        return new ItemStack(Items.CAMPFIRE);
    }

    @Override
    public boolean canBeFinale(GameStartContext context) {
        return false;
    }

    @Override
    public boolean canBePlayed(GameStartContext context) {
        return context.getParticipantCount() >= 2;
    }

    @Override
    public MiniGameInstance createInstance(MiniGameHandle gameHandle) {
        return new CozyCampfireInstance(gameHandle);
    }
}

File: src/minigames/cozy_campfire/src/main/kotlin/work/lclpnet/ap2/game/cozy_campfire/CozyCampfireMiniGame.kt
package work.lclpnet.ap2.game.cozy_campfire

import net.minecraft.core.RegistryAccess
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.Items
import work.lclpnet.ap2.ApConstants
import work.lclpnet.ap2.api.base.GameStartContext
import work.lclpnet.ap2.game.GameType
import work.lclpnet.ap2.game.MiniGame
import work.lclpnet.ap2.game.MiniGameFactory

private const val DEBUG_PLAYER_CONSTRAINT = false

class CozyCampfireMiniGame : MiniGame {
    override val id = ApConstants.identifier("cozy_campfire")
    override val type = GameType.TEAM
    override val author = ApConstants.PERSON_LCLP
    override fun getIcon(manager: RegistryAccess): ItemStack = ItemStack(Items.CAMPFIRE)
    override fun canBeFinale(context: GameStartContext): Boolean = false

    override fun canBePlayed(context: GameStartContext): Boolean {
        val count = context.participantCount
        if (DEBUG_PLAYER_CONSTRAINT) return count >= 2
        return count == 2 || count >= 4
    }

    override fun createFactory(): MiniGameFactory = CozyCampfireFactory()
}
